File: proj/httpcachez-java/src/test/java/com/murerz/httpcachez/HttpServer.java
package com.murerz.httpcachez;

import java.io.Closeable;

public class HttpServer implements Closeable {

	private static final HttpServer ME = new HttpServer();

	public static HttpServer me() {
		return ME;
	}

	public HttpServer start(int port) {
		return this;
	}

	public int getPort() {
		return 0;
	}

	@Override
	public void close() {
	}

}

File: proj/darkproxy/src/http/HttpServer.js
const LOG = require('../Logger.js')(__filename);
const http = require('http');

class HttpServer {

  constructor(params) {
    this.params = params;
  }

  toString() {
    return `[${this.constructor.name} ${this.port}]`;
  }

  start() {
    return new Promise((resolve, reject) => {
      this.conns = {};
      this.server = http.createServer();
      this.server.listen(this.params, () => {
        this.server.removeListener('error', reject);
        this.port = this.server.address().port;
        LOG.info(`${this} started`);
        resolve();
      });
      this.server.on('error', reject);
      this.server.on('request', (req, resp) => {
      });
    });
  }

  stop() {
    return new Promise((resolve, reject) => {
      this.server.close(() => {
        LOG.info(`${this} stopped`);
        resolve();
      });
    });
  }

}

exports.HttpServer = HttpServer;
